src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj

```diff
@@ -77,39 +77,60 @@
   (log/debug (i18n/trs "No handler defined for websocket event ''{0}'' with args: ''{1}''"
                        event args)))
 
+(def client-count (atom 0))
+(defn extract-CN-from-certs
+  [x509certs]
+  (.getSubjectX500Principal (first x509certs)))
+
 (schema/defn ^:always-validate proxy-ws-adapter :- WebSocketAdapter
   [handlers :- WebsocketHandlers
    x509certs :- [X509Certificate]
    requestPath :- String
    closureLatch :- CountDownLatch]
-  (let [{:keys [on-connect on-error on-text on-close on-bytes]
+  (let [client-id (swap! client-count inc)
+        {:keys [on-connect on-error on-text on-close on-bytes]
          :or {on-connect (partial no-handler :on-connect)
               on-error   (partial no-handler :on-error)
               on-text    (partial no-handler :on-text)
               on-close   (partial no-handler :on-close)
               on-bytes   (partial no-handler :on-bytes)}} handlers]
     (proxy [WebSocketAdapter CertGetter ClosureLatchSyncer] []
       (onWebSocketConnect [^Session session]
+        (log/tracef "%d on-connect certname:%s uri:%s" client-id (extract-CN-from-certs x509certs) requestPath)
```

Comment:
Can you define the value of `(extract-CN-from-certs x509certs)` in the let block so we don't have to do the work on every log line?

Reply:
That way it is done once per instance, rather than once per log line.

Reply:
Whoops yeah

Reply:
Updated